src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -33,17 +34,19 @@ class Router extends BaseRouter implements WarmableInterface, ServiceSubscriberI
     private $collectedParameters = array();
 
     /**
-     * @param ContainerInterface $container A ContainerInterface instance
-     * @param mixed              $resource  The main resource to load
-     * @param array              $options   An array of options
-     * @param RequestContext     $context   The context
+     * @param ContainerInterface   $container A ContainerInterface instance
+     * @param mixed                $resource  The main resource to load
+     * @param array                $options   An array of options
+     * @param RequestContext       $context   The context
+     * @param LoggerInterface|null $logger
      */
-    public function __construct(ContainerInterface $container, $resource, array $options = array(), RequestContext $context = null)
+    public function __construct(ContainerInterface $container, $resource, array $options = array(), RequestContext $context = null, LoggerInterface $logger = null)
```

Comment:
how about using a nullable typehint ?

Reply:
We reverted it last time we tried adding one 😄 
See discussion starting from https://github.com/symfony/symfony/pull/22743#discussion_r117262044